### ISSUE TITLE
Moving the waveglow output to CPU is very slow

### ISSUE THREAD

**Issue by CorentinJ** [NVIDIA/tacotron2#140]
In the [inference demo](https://github.com/NVIDIA/tacotron2/blob/master/inference.ipynb), after these lines:
```
with torch.no_grad():
    audio = waveglow.infer(mel_outputs_postnet, sigma=0.666)
```

I call `audio.detach().cpu()` and `cpu()` alone takes ~6 seconds to run for a tensor of approximately 60K elements. Is that expected behaviour from torch or is there something funky going on with the waveglow output tensor?

**Comment by CorentinJ** [comment]
Alright my bad, it's just that the operations are asynchronous. Calling `torch.cuda.synchronize()` between each major step of the synthesis helped isolate the bottleneck. It takes ~8 seconds to generate the waveform of a short sentence on my GTX 1080. That doesn't seem right, considering this quote from the Waveglow paper:

> Despite the simplicity of the model, our PyTorch implementation synthesizes speech at more than 500kHz on an NVIDIA V100 GPU: more than 25 times faster than real time

**Comment by rafaelvalle** [comment]
Our performance results were computed on a NVIDIA V100 with large input size, i.e. ~ 10 seconds or 880 mels.
